src/main/java/teammates/logic/core/CoursesLogic.java

```diff
@@ -182,22 +184,39 @@ public CourseDetailsBundle getCourseDetails(String courseId) throws EntityDoesNo
     }
 
     public List<String> getSectionsNameForCourse(String courseId) throws EntityDoesNotExistException {
+        return getSectionsNameForCourse(courseId, true);  
+    }
 
-        verifyCourseIsPresent(courseId);
-        
+    public List<String> getSectionsNameForCourse(CourseAttributes course) throws EntityDoesNotExistException {
+        Assumption.assertNotNull("Course is null", course);
+        return getSectionsNameForCourse(course.id, false);
+    }
+    
+    /**
+     * Get list of section names for a course with or without a need to check if the course is existent
+     * @param courseId Course ID of the course
+     * @param hasCheckIsPresent Determine whether it is necessary to check if the course exists
+     * @return list of sections names from the specified course
+     * @throws EntityDoesNotExistException
+     */
+    private List<String> getSectionsNameForCourse(String courseId, boolean hasCheckIsPresent) 
```

Comment:
Change variable name to isCourseVerified? And verifyCourseIsPresent only if it is false.